controllers/email-l10n.js

```diff
@@ -3,154 +3,138 @@
-    return res.redirect('/email-l10n')
-  }
-
-  const unsafeBreachesForEmail = [];
-  ['Dropbox', 'Apollo', 'Adobe'].forEach(name => {
-    unsafeBreachesForEmail.push(req.app.locals.breaches.filter(breach => breach.Name === name)[0])
-  })
-
-  const emailContent = ((req) => {
-    switch (req.query.type) {
-      case 'pre-fxa':
-        return {
-          emailSubject: req.fluentFormat('pre-fxa-subject'),
-          preFxaEmail: true,
-          breachAlert: null
-        }
-      case 'noBreaches':
-        return {
-          emailSubject: req.fluentFormat('email-subject-no-breaches'),
-          breachAlert: null,
-          unsafeBreachesForEmail: []
-        }
-      case 'breachAlert':
-        return {
-          emailSubject: req.fluentFormat('breach-alert-subject'),
-          breachAlert: req.app.locals.breaches.filter(breach => breach.Name === 'LinkedIn')[0],
-          unsafeBreachesForEmail: null,
-          preFxaSubscriber: true
-        }
-      case 'email_verify':
-        return {
-          emailSubject: req.fluentFormat('email-subject-verify'),
-          breachAlert: null,
-          unsafeBreachesForEmail: null
-        }
-      case 'multipleBreaches':
-        return {
-          emailSubject: req.fluentFormat('email-subject-found-breaches'),
-          unsafeBreachesForEmail,
-          breachAlert: null
-        }
-      default:
-        return {
-          emailSubject: req.fluentFormat('email-subject-found-breaches'),
-          unsafeBreachesForEmail: unsafeBreachesForEmail.slice(0, 1),
-          breachAlert: null
-        }
-    }
-  })(req)
+  const emailStr = await readFile(path.resolve('views/layouts/email-2022.hbs'), { encoding: 'utf-8' })
+  const emailStyle = emailStr.substring(emailStr.indexOf('<style>'), emailStr.indexOf('</style>') + 8)
+  const { unsafeBreachesForEmail, emailSubject, breachAlert, unsubscribeUrl, heading, subheading } = getPartialData(req.query.type, req.app.locals.breaches)
 
   res.render('email_l10n', {
     layout: 'email_l10n_mockups.hbs',
-    unsafeBreachesForEmail: emailContent.unsafeBreachesForEmail,
+    unsafeBreachesForEmail,
     supportedLocales: req.supportedLocales,
     whichPartial: `email_partials/${req.query.partial}`,
-    breachedEmail: 'breachedEmail@testing.com',
-    recipientEmail: 'recipientEmail@testing.com',
-    breachAlert: emailContent.breachAlert,
-    emailSubject: emailContent.emailSubject,
-    preFxaSubscriber: emailContent.preFxaSubscriber,
-    email,
-    preFxaEmail: emailContent.preFxaEmail,
-    ctaHref: EmailUtils.getEmailCtaHref('breach-alert', 'go-to-dashboard')
+    partialType: req.query.type,
+    breachedEmail: req.user?.primary_email || 'breachedEmail@testing.com',
+    breachAlert,
+    emailSubject: req.fluentFormat(emailSubject),
+    heading: req.fluentFormat(heading),
+    subheading: req.fluentFormat(subheading),
+    ctaHref: EmailUtils.getEmailCtaHref(req.query.type, 'dashboard-cta'),
```

Comment:
yeah... I believe this is just for the mockup though.  The link ends up being `http://localhost:6060/?utm_source=fx-monitor&utm_medium=email&utm_campaign=email_verify&utm_content=dashboard-cta`, which I figured was fine since we don't _really_ want to verify (or unsubscribe) a user.

Not sure how you're getting an undefined param in your link?

Reply:
I'm OK leaving a dashboard link for the mockup. I think that sets my expectations for these pages - layout is correct, but links and `utm_campaign` can be subtly different.

It was two weeks ago, but I believe it was from manually calling `EmailUtils.getVerificationUrl()` without a request user. In the test monthly breach email, I do get a valid token in the `Unsubscribe` link.